oneview_redfish_toolkit/api/redfish_json_validator.py

```diff
@@ -96,3 +96,36 @@ def serialize(self):
             default=lambda o: o.__dict__,
             sort_keys=False,
             indent=indent)
+
+    def get_resource_by_id(self, resource_list, resource_number_key,
+                           resource_id, resource_name):
+        """Gets a specific resource in the resource list
+
+            Validates the resource ID and gets the resource in
+            the resource list.
+
+            Args:
+                resource_list: List of resources.
+                resource_number_key: Field name of the resource
+                    number in the JSON.
+                resource_id: Resource's ID that will be searched
+                    in the resource list.
+                resource_name: Resource name used to improve error message.
+
+            Returns:
+                Resource in the list.
+                Returns None if the resource was not found.
+
+            Exception:
+                OneViewRedfishError: If the ID is not an integer.
+        """
+        try:
+            resource_id = int(resource_id)
+        except ValueError:
+            raise OneViewRedfishError("Invalid {} ID".format(resource_name))
+
+        for resource in resource_list:
+            if resource[resource_number_key] == resource_id:
+                return resource
+
+        return None
```

Comment:
this is not acceptable. When this function is called it MUST!!!! return a valid resource or should raise an exception.

Reply:
Fixed.